modules/browser-node/src/channelProvider.ts

```diff
@@ -0,0 +1,280 @@
+import EventEmitter from "eventemitter3";
+import {
+  ChannelRpcMethod,
+  EngineEvent,
+  ChannelRpcMethods,
+  EngineEvents,
+  EngineEventMap,
+  IVectorEngine,
+  EngineParams,
+} from "@connext/vector-types";
+import { constructRpcRequest, safeJsonParse } from "@connext/vector-utils";
+
+export function isEventName(event: string): event is EngineEvent {
+  return event in EngineEvents;
+}
+
+export function isMethodName(event: string): event is ChannelRpcMethod {
+  return event in ChannelRpcMethods;
+}
+
+export interface IframeOptions {
+  id: string;
+  src: string;
+}
+
+export function payloadId(): number {
+  const date = new Date().getTime() * Math.pow(10, 3);
+  const extra = Math.floor(Math.random() * Math.pow(10, 3));
+  return date + extra;
+}
+
+export function renderElement(name: string, attr: any, target: HTMLElement): HTMLElement {
+  const elm = document.createElement(name);
+  Object.keys(attr).forEach(key => {
+    elm[key] = attr[key];
+  });
+  target.appendChild(elm);
+  return elm;
+}
+
+export interface IRpcChannelProvider {
+  connected: boolean;
+  send(payload: EngineParams.RpcRequest): Promise<any>;
+  open(): Promise<void>;
+  close(): Promise<void>;
+  on<T extends EngineEvent>(
+    event: T,
+    callback: (payload: EngineEventMap[T]) => void | Promise<void>,
+    filter?: (payload: EngineEventMap[T]) => boolean,
+  ): Promise<void>;
+  once<T extends EngineEvent>(
+    event: T,
+    callback: (payload: EngineEventMap[T]) => void | Promise<void>,
+    filter?: (payload: EngineEventMap[T]) => boolean,
+  ): Promise<void>;
+}
```

Comment:
This seems like something that should go into the `types` module

Reply:
It's only relevant to the browser-node. I chatted about this with @bohendo too, we agree that only shared types should go into the types module.